core/src/main/java/hudson/model/UpdateCenter.java

```diff
@@ -1502,23 +1502,38 @@ public Plugin getPlugin() {
         @Override
         public void run() {
             try {
-                plugin.getInstalled().enable();
-            } catch (IOException e) {
-                LOGGER.log(Level.SEVERE, "Failed to enable " + plugin.getDisplayName(), e);
-                error = e;
-            }
-            
-            if (dynamicLoad) {
-                try {
-                    // remove the existing, disabled inactive plugin to force a new one to load
-                    pm.dynamicLoad(getDestination(), true);
-                } catch (Exception e) {
-                    LOGGER.log(Level.SEVERE, "Failed to dynamically load " + plugin.getDisplayName(), e);
-                    error = e;
-                    requiresRestart = true;
+                PluginWrapper installed = plugin.getInstalled();
+                synchronized(installed) {
+                    if (!installed.isEnabled()) {
+                        try {
+                            installed.enable();
+                        } catch (IOException e) {
+                            LOGGER.log(Level.SEVERE, "Failed to enable " + plugin.getDisplayName(), e);
+                            error = e;
+                            status = new Failure(e);
```

Comment:
For this case it may help if you use a local <code>final Status _status</code> declaration. It will allow to ensure you set it once and only once in every code branch. Then just assign it at the end